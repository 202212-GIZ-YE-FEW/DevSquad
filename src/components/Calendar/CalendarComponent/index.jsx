import React from "react";
import { useState } from "react";
export default function CalendarComponent({ day, year, month }) {
    const [isChecked, setIsChecked] = useState(false);
    let selectedDate = [];
    function handleDayClick() {
        selectedDate.push(`${year}-${month}-${day}`);
        console.log(selectedDate);
    }
    return (
        <label>
            <input
                type='checkbox'
                key={day}
                onChange={() => setIsChecked((isChecked) => !isChecked)}
                hidden
                onClick={handleDayClick}
            />
            <span
                className={
                    isChecked
                        ? "bg-primary-orange rounded font-bold text-lg p-1"
                        : "bg-white rounded font-bold text-lg"
                }
            >
                {day}
            </span>
        </label>
    );
}
