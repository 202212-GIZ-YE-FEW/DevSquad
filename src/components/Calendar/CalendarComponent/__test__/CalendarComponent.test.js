import renderer from "react-test-renderer";

import CalendarComponent from "../index";

it("renders correctly", () => {
    const tree = renderer.create(<CalendarComponent />).toJSON();
    expect(tree).toMatchSnapshot();
});
