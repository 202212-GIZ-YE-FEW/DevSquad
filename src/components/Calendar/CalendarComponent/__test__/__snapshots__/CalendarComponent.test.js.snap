// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label>
  <input
    hidden={true}
    onChange={[Function]}
    onClick={[Function]}
    type="checkbox"
  />
  <span
    className="bg-white rounded font-bold text-lg"
  />
</label>
`;
